boa/src/exec/mod.rs

```diff
@@ -420,11 +420,24 @@ impl Executor for Interpreter {
                         | Node::FunctionExpr(_, _, _)
                         | Node::New(_)
                         | Node::Object(_)
-                        | Node::TypeOf(_)
                         | Node::UnaryOp(_, _) => Value::boolean(true),
                         _ => panic!("SyntaxError: wrong delete argument {}", node),
                     },
-                    _ => unimplemented!(),
+                    UnaryOp::TypeOf => Value::from(match v_a.data() {
+                        ValueData::Undefined => "undefined",
+                        ValueData::Symbol(_) => "symbol",
+                        ValueData::Null => "object",
+                        ValueData::Boolean(_) => "boolean",
+                        ValueData::Rational(_) | ValueData::Integer(_) => "number",
+                        ValueData::String(_) => "string",
+                        ValueData::Object(ref o) => {
+                            if o.deref().borrow().is_callable() {
+                                "function"
+                            } else {
+                                "object"
+                            }
+                        }
+                    }),
```

Comment:
Why do we do this here when we can just call `value.get_type()` which returns a `&'static str` it seems like were duping code.

Reply:
I think i mentioned in the original issue that we don't need both https://github.com/jasonwilliams/boa/issues/359#issue-610887558

Reply:
as @HalidOdat said maybe we can just call get_type() from here